Formula/amp.rb

```diff
@@ -0,0 +1,37 @@
+class Amp < Formula
+  desc "Text editor for your terminal"
+  homepage "https://amp.rs"
+  url "https://github.com/jmacdonald/amp/archive/0.3.2.tar.gz"
+  sha256 "9080c894111149d58a0b805e0c08edaf1d9e7cd56c4b955ba9111feec6b55208"
+  head "https://github.com/jmacdonald/amp.git"
+
+  depends_on "rust" => :build
+  depends_on "cmake" => :build
+  depends_on "openssl"
+  depends_on "libssh2"
+
+  def install
+    system "cargo", "build", "--release"
+    bin.install "target/release/amp"
+  end
+
+  test do
+    (testpath/"test.exp").write <<~EOS
+      spawn "#{bin}/amp" "#{testpath}/test_write"
```

Comment:
Looks like it's expecting this file to already exist?
```
No such file or directory @ rb_sysopen - /tmp/amp-test-20180204-98058-kbm6ck/test_write
```
Might be worth `touch`ing it as the first line of the test block.